compute_horde/compute_horde/miner_client/base.py

```diff
@@ -59,12 +55,15 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
 
         await self.transport.stop()
 
-    async def send_model(self, model: BaseRequest):
+    async def send_model(self, model: BaseRequest, error_event_callback=None):
         while True:
             try:
                 await self.transport.send(model.model_dump_json())
             except TransportConnectionError as ex:
-                logger.error(f"Could not send to miner {self.miner_name}: {str(ex)}")
+                msg = f"Could not send to miner {self.miner_name}: {str(ex)}"
+                logger.warning(msg)
+                if error_event_callback:
+                    await error_event_callback(msg)
```

Comment:
```suggestion
                    try:
                        await error_event_callback(msg)
                    except Exception as callback_ex:
                        logger.error("Could not execute error event callback: %s", str(callback_ex))
```

So that it doesn't affect the loop execution